Odata-docs/webapi-8/fundamentals/overview.md

```diff
@@ -0,0 +1,582 @@
+
+Visit the [routing documentation](/odata/webapi-8/fundamentals/routing-overview) to learn more about OData routing.
+
+## Query handling
+
+OData defines various query options that clients can add to an OData request URL to transform or limit the query results. Here are some of the common query options:
+- `$filter` allows you to filter results based on specific conditions: `/Products?$filter=Price lt 1000 and Category eq 'Electronics'` will only return electronic products whose price is less than 1,000.
+- `$select` allows you to limit the fields that are included in the response: `/Products?$select=Name,Category,DateCreated` will only include the `Name`, `Category`, `DateCreated` fields in the response
+- `$expand` allows you to join and include related data in the response: `/OrderItems?$expand=Product` will return order items and include the related product in the response
+- `$skip` and `$top` allow you to limit the number of records in the response and can be used for client-driven pagination: `/Products$top=10&$skip=20` returns the third page of 10 products.
+- `$orderby` allows you to sort the results: `/Products?$orderby=DateCreated desc` will sort the products by date in descending order (most recent first).
+- `$apply` allows you to apply aggregations: `/Orders?$apply=groupby((Product/Name), aggregate($count as Count))` returns the number of orders per product name.
+
+These query options are not enabled by default. You can enable and configure them individually through the options passed to the `AddOData()` method:
+
+```c#
+services.AddControllers().AddOData(
+    options => options.Select().Filter().Expand().OrderBy().SetMaxTop(100).AddRouteComponents(edmModel));
+```
+In this example, we only enable the `$select`, `$filter`, `$expand`, `$orderby` and `$top` query options and we're setting the max allowable value for `$top` to be 100. If the client tries to use an OData query option that's not enabled, the service will return an error.
+
+You can enable all supported query options in one go using the `EnableQueryFeatures` method:
+
+```c#
+services.AddControllers().AddOData(
+    options => options.EnableQueryFeatures(100).AddRouteComponents(edmModel));
+```
+
+The argument passed to `EnableQueryFeatures` is the max top value. You can set it to `null` if you don't want to impose a limit the value specified in the `$top` query option.
+
+After enabling query options on the service, you also need to enable querying on specific endpoints where you want to support query options. You can do this by adding the [`[EnableQuery]`](/dotnet/api/microsoft.aspnetcore.odata.query.enablequeryattribute) attribute to the corresponding controller actions. The [`EnableQuery`] attribute automatically applies the query options to the results returned by the controller action.
```

Comment:
```suggestion
After enabling query options on the service, you also need to enable querying on specific endpoints where you want to support query options. You can do this by adding the [`EnableQuery`](/dotnet/api/microsoft.aspnetcore.odata.query.enablequeryattribute) attribute to the corresponding controller actions. The `EnableQuery` attribute automatically applies the query options to the results returned by the controller action.
```

I don't think you need to include the square brackets every time you're referring to a method attribute

Reply:
Would it be clear that it's an attribute? Or should I call it `EnableQueryAttribute`?